tools/artifact_gen/MODULE.bazel

```diff
@@ -0,0 +1,30 @@
+# Copyright 2025 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+bazel_dep(name = "pugixml", version = "1.15")
```

Comment:
I'd suggest using `dev_dependency=True` for all dependencies other than `abseil-cpp`. And please make sure that users of released version don't need to generate project files when they just use gRPC.